bb-custom-attributes.php

```diff
@@ -105,13 +129,50 @@ public function filterAdvancedModule($form, $id)
      *
      * @return array
      */
-    public function filterAttributes($attributes, $module)
+    public function filterModuleAttributes($attributes, $module)
     {
         if (isset($module->settings->custom_attributes)) {
             foreach ($module->settings->custom_attributes as $attribute) {
                 $key = esc_attr($attribute->key);
                 if ('yes' === $attribute->override || ! isset($attributes[$key])) {
-                    $attributes[$key] = esc_attr($attribute->value);
+                    $value = do_shortcode(esc_attr($attribute->value));
+                    $attributes[$key] = $value;
+                }
+            }
+        }
+
+        return $attributes;
+    }
+    
+    /**
+	 * Adds the custom attributes to the column being rendered
+	 */
+    public function filterColAttributes($attributes, $col)
+    {
+        if (isset($col->settings->custom_attributes)) {
+            foreach ($col->settings->custom_attributes as $attribute) {
+                $key = esc_attr($attribute->key);
+                if ('yes' === $attribute->override || ! isset($attributes[$key])) {
+                    $value = do_shortcode(esc_attr($attribute->value));
+                    $attributes[$key] = $value;
+                }
+            }
+        }
+
+        return $attributes;
+    }
+	
+	/**
+	 * Adds the custom attributes to the row being rendered
+	 */
+    public function filterRowAttributes($attributes, $row)
+    {
+        if (isset($row->settings->custom_attributes)) {
+            foreach ($row->settings->custom_attributes as $attribute) {
+                $key = esc_attr($attribute->key);
+                if ('yes' === $attribute->override || ! isset($attributes[$key])) {
+                    $value = do_shortcode(esc_attr($attribute->value));
+                    $attributes[$key] = $value;
                 }
             }
```

Comment:
Do these need to be three separate filter functions? Unless I'm missing something the only difference is the name of the second parameter.

Reply:
You are correct. I didn't even notice they were the same! I'll merge them